### PR TITLE
add local storage  of form data

### PR DESCRIPTION
**Pull request by Fourteen98**
New Features:

- When the user changes the content of any input field, the data is saved to the local storage.
- When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data.

### CODE REVIEW

**Comment by Fourteen98** [comment]
Dear Code Reviewer, 
I am working solo  this week for all the task.
image

File: js/index.js [hunk 1]
```diff
@@ -52,22 +52,57 @@ closePopupBtn.addEventListener('click', () => {
...
+  const toString = JSON.stringify(formData);
+  localStorage.setItem('formData', toString);
+  setData();
+}
+
+if (!localStorage.getItem('formData')) {
+  getData();
+} else {
+  setData();
+}
```
Comment (Mre55) [inline js/index.js]: Please add `onchange` attribute for your HTML form input fields that way you can save every change that the user inputs and when the page reloads it'll get those values from `local storage` and file them out. Here when I input values but before submitting when I refresh the page it's not getting those values kindly fix that.

https://user-images.githubusercontent.com/34578631/174427283-db323815-59bc-4936-a01c-71f5b41acd91.mov


Also you can check this if statement either on `window.onload = () => {...}` or on `document.addEventListener('DOMContentLoaded', () => {...})` to get the values.

**Review by Mre55 — CHANGES_REQUESTED** [review]
**Hi @Fourteen98,**

You have done a great job so far! 💪 👏 
You still need to work on a few issues before moving on to the next project, but you're almost there!

## To highlight 🎯

- The Pull Request is following the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- The commit messages are meaningful ✅ 
- The README is well documented ✅ 
- Linters are working properly ✅ 

## Required Changes ♻️🛠️ 
_Check the comments under the review._





## Optional suggestions 🟡 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: js/index.js [hunk 2]
```diff
@@ -52,22 +52,59 @@ closePopupBtn.addEventListener('click', () => {
...
+  localStorage.setItem('formData', toStringData);
+
+  setData();
+}
+
+if (!localStorage.getItem('formData')) {
+  saveData();
+} else {
+  setData();
+}
```
Comment (ishimwezachee) [inline js/index.js]: - [ ]  Please implement the feedback your previous reviewer suggested, Your form should save the data in localstorage, and the last submitted data should persist in your form.  I am trying to submit the form, but the data are not being saved in localstorage.

**Review by ishimwezachee — CHANGES_REQUESTED** [review]
Hi @Fourteen98   , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


### To highlight 💯 
- Good commit message ✔️ 
- No Linter errors ✔️ 
- Follow Github flow ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on @Ishimwe Zachee  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: js/index.js (hunk 1 above)
Comment (Fourteen98) [reply js/index.js]: > Please add `onchange` attribute for your HTML form input fields that way you can save every change that the user inputs and when the page reloads it'll get those values from `local storage` and file them out. Here when I input values but before submitting when I refresh the page it's not getting those values kindly fix that.
> 
>  Screen.Recording.2022-06-18.at.10.07.10.AM.mov 
> 
> Also you can check this if statement either on `window.onload = () => {...}` or on `document.addEventListener('DOMContentLoaded', () => {...})` to get the values.

@Mre55 , been trying to implement. but still not working   well for me. Can you kindly help me out here. Left with this task so i finish my week.

File: js/index.js (hunk 2 above)
Comment (Fourteen98) [reply js/index.js]: > [ ] Please implement the feedback your previous reviewer suggested, Your form should save the data in localstorage, and the last submitted data should persist in your form. I am trying to submit the form, but the data are not being saved in localstorage.

@ishimwezachee Been trying to and i am still not  getting it. I am just left with this task. can you please describe in  more details.

File: js/index.js (hunk 1 above)
Comment (Mre55) [reply js/index.js]: @Fourteen98, sorry for the late response have you manage to fix it or should I help?

Comment (Fourteen98) [reply js/index.js]: > @Fourteen98, sorry for the late response have you manage to fix it or should I help?

No, Still having some issues. @Mre55

Comment (Fourteen98) [reply js/index.js]: I was Delayed on  this pair programming task by my coding partner untill she was removed. Working solo is hard @Mre55

Comment (Mre55) [reply js/index.js]: Let me what you have until now.

Comment (Mre55) [reply js/index.js]: @Fourteen98, can we join a call?

Comment (Fourteen98) [reply js/index.js]: @Mre55  sure

Comment (Fourteen98) [reply js/index.js]: > @Mre55 sure

will be glad

Comment (Fourteen98) [reply js/index.js]: https://microverse.zoom.us/j/93685431683?pwd=dG8zRmZ2MklCSGZYeUZJdktVaGZUUT09
@Mre55

Comment (Fourteen98) [reply js/index.js]: @Mre55 ,i think it is working now, can you kindly confirm that.

Comment (Mre55) [reply js/index.js]: @Fourteen98, yes it's working fine you can submit now.

Comment (Fourteen98) [reply js/index.js]: W

> @Fourteen98, yes it's working fine you can submit now.

@Mre55  thanks so much,

Comment (Mre55) [reply js/index.js]: @Fourteen98, you're welcome.

File: js/index.js [hunk 3]
```diff
@@ -71,3 +71,23 @@ form.addEventListener('submit', (e) => {
...
+
+const formElements = document.querySelectorAll('input', 'area');
+formElements.forEach((element) => {
+  element.value = formData[element.name];
+  element.addEventListener('input', (e) => {
+    formData[e.target.name] = e.target.value;
+    localStorage.setItem('formData', JSON.stringify(formData));
+  });
+});
```
Comment (sja-thedude) [inline js/index.js]: - [ ] Great job so far, I tried searching your slack id to contact you but couldn't find it. Your name and email are working correctly with persistent data. however, your message is not persistent upon refreshed. Kindly refer below for the same and fix it.

https://user-images.githubusercontent.com/70269507/174439572-fedc06fc-2e03-416f-980e-d95df0327aa2.mov

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @Fourteen98, This is @sja-thedude your code reviewer.

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

200w

 ## Highlights
✔️  Good job so far

 ## Required Changes ♻️
- [ ] _Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: js/index.js (hunk 3 above)
Comment (Fourteen98) [reply js/index.js]: @sja-thedude 
slack name: Muhyideen Elias

Comment (Fourteen98) [reply js/index.js]: @sja-thedude
email : emuhyideen6@gmail.com

File: js/index.js (hunk 1 above)
Comment (Fourteen98) [reply js/index.js]: @Mre55 , i was turned down. says  the message has the same problem.

Comment (Fourteen98) [reply js/index.js]: @Mre55 i have reached out on slack

Comment (Mre55) [reply js/index.js]: @Fourteen98, I have replayed on slack

**Review by keroloussamy — APPROVED** [review]
Hi @Fourteen98, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To highlight 🎯

- No linter errors.✔️
- Good code design and implementation. ✔️
- Add all the requirements. ✔️

clap



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**